drivers/gpio/gpio_rt1718s_alert.c

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT richtek_rt1718s_alert
+
+/**
+ * @file Driver for handling RS1718S chip alert.
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/gpio/rt1718s.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/util_macro.h>
+LOG_MODULE_DECLARE(gpio_rt1718s_alert, CONFIG_GPIO_LOG_LEVEL);
+
+/* Driver config */
+struct rt1718s_alert_config {
+	/* Alert GPIO pin */
+	const struct gpio_dt_spec irq_gpio;
+	/* RT1718S devices which share the same alert pin */
+	const struct device **rt1718s_dev;
+	/* Number of RT1718S devices on the alert pin */
+	uint32_t rt1718s_num;
+};
+
+/* Driver data */
+struct rt1718s_alert_data {
+	/* Alert handler device */
+	const struct device *alert_dev;
+	/* Alert pin callback */
+	struct gpio_callback gpio_cb;
+	/* Alert worker */
+	struct k_work alert_worker;
+};
+
+static void rt1718s_alert_callback(const struct device *dev, struct gpio_callback *cb,
+				   uint32_t pins)
+{
+	ARG_UNUSED(pins);
+	struct rt1718s_alert_data *data = CONTAINER_OF(cb, struct rt1718s_alert_data, gpio_cb);
+
+	k_work_submit(&data->alert_worker);
+}
+
+static void rt1718s_alert_worker(struct k_work *work)
```

Comment:
Is the `richtek,rt1718s-alert`` driver actually needed?  I see that it is provided so that multiple RT1718s devices can share the same alert GPIO connected to the SoC.  However, the GPIO API already supports chaining together multiple callbacks.  Is it possible to move the `irq-gpios` property into the parent device `richtek,rt1718s` directly?

Reply:
Good point. Multiple callbacks per one GPIO + `irq-gpios` property in the parent node works well here.